samza-kafka/src/main/scala/org/apache/samza/system/kafka/BrokerProxy.scala

```diff
@@ -181,7 +181,7 @@ class BrokerProxy(
       firstCallBarrier.countDown()
 
       // Split response into errors and non errors, processing the errors first
-      val (nonErrorResponses, errorResponses) = response.data.entrySet().partition(_.getValue.error == ErrorMapping.NoError)
+      val (nonErrorResponses, errorResponses) = response.data.toMap.entrySet().partition(_.getValue.error == ErrorMapping.NoError)
```

Comment:
Is the structure of `response` different between 0.10.0.1 and 0.10.2.0? 

Also, are there bug fixes in 0.10.2.0 that you require?